platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/PlatformStateAccessorSingleton.java

```diff
@@ -0,0 +1,443 @@
+import com.hedera.hapi.node.base.Timestamp;
+import com.hedera.hapi.platform.state.MinimumJudgeInfo;
+import com.hedera.hapi.platform.state.PlatformState;
+import com.hedera.hapi.platform.state.PlatformState.Builder;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.merkle.MerkleNode;
+import com.swirlds.platform.consensus.ConsensusSnapshot;
+import com.swirlds.platform.event.AncientMode;
+import com.swirlds.platform.state.schemas.V0540PlatformStateSchema;
+import com.swirlds.platform.system.SoftwareVersion;
+import com.swirlds.platform.system.address.AddressBook;
+import com.swirlds.state.merkle.StateMetadata;
+import com.swirlds.state.merkle.singleton.SingletonNode;
+import com.swirlds.state.spi.CommittableWritableStates;
+import com.swirlds.state.spi.ReadableSingletonState;
+import com.swirlds.state.spi.WritableSingletonState;
+import com.swirlds.state.spi.WritableStates;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.time.Instant;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * This class is a drop-in replacement for {@link PlatformState} that uses singleton object under the hood.`
+ * It assumes that the instance of {@link MerkleRoot} passed in the constructor is a {@link MerkleStateRoot} instance.
+ * It registers the singleton object for the platform state in the {@link MerkleStateRoot} if it is not already present.
+ */
+public class PlatformStateAccessorSingleton implements PlatformStateAccessor {
+
+    private final MerkleStateRoot root;
+    private final StateMetadata<String, PlatformState> md;
+    private final Function<SemanticVersion, SoftwareVersion> softwareVersionSupplier;
+    private final Supplier<SingletonNode<PlatformState>> platformSingletonSupplier;
+
+    /**
+     * Create a new instance of the platform state accessor.
+     *
+     * @param root Merkle state root
+     */
+    public PlatformStateAccessorSingleton(@NonNull MerkleStateRoot root) {
+        this.root = root;
+        final Function<SemanticVersion, SoftwareVersion> softwareVersionSupplier = root.getSoftwareVersionSupplier();
+        this.softwareVersionSupplier = requireNonNull(softwareVersionSupplier);
+        V0540PlatformStateSchema schema = new V0540PlatformStateSchema();
+        md = new StateMetadata<>(
+                PlatformStateAccessor.PLATFORM_NAME,
+                schema,
+                schema.statesToCreate().iterator().next());
+        platformSingletonSupplier = () -> new SingletonNode<>(
+                md.serviceName(),
+                md.stateDefinition().stateKey(),
+                md.singletonClassId(),
+                md.stateDefinition().valueCodec(),
+                PlatformState.newBuilder().build());
+
+        root.putServiceStateIfAbsent(md, platformSingletonSupplier);
```

Comment:
I can't say for sure, but it doesn't look right. Any schema should be registered just once. In particular, this new `V0540PlatformStateSchema` is already registered in `Hedera`, and schema registration should have triggered platform state creation. Why is it needed here?

Reply:
No longer relevant.